TESTS/netsocket/ethernet.json

```diff
@@ -0,0 +1,27 @@
+{
+    "config": {
+        "header-file": {
+            "help" : "String for including your driver header file",
+            "value" : "\"EthernetInterface.h\""
+        },
+        "object-construction" : {
```

Comment:
what's the goal of this two `object-construction` and `connect-statement` ? Replacing the code like this via preprocessor is not ideal. I would avoid it if we could

Using the common implementation that would receive  the object to be used via a parameter or something similar? 

No strong preference